benchmark/all/all_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package all runs all the benchmark tests of Finch.
```

Comment:
Does this need to be its own package? Can't the BenchmarkAll function just be in the benchmark package?

Reply:
I was also thinking this alternative and finally chose having a separated package. Here is my thought process.

BenchmarkAll() can't call BenchmarkVM() and BenchmarkContainer() in current logic because they have to be separated testing.B interface. BenchmarkAll() calls suite functions similar to what BenchmarkVM() and BenchmarkContainer(). So even logically, BenchmarkAll() seems to be upper layer of BenchmarkVM() and BenchmarkContainer(), they are actually in same layer in current implementation.

If BenchmarkAll() is in upper layer, I think it makes more sense to put in benchmark package. Now they are in same layer, it might be cleaner to treat them as 3 packages in parallel, to match the three make targets in makefile.

Benchmark package can also be cleaner, which only has the util functions instead of testing.B functions.

Reply:
Ok, that reasoning makes sense to me, thanks for the details 👍